CMake/external_libcurl.cmake

```diff
@@ -37,13 +44,17 @@ if(CHECK_FOR_UPDATES)
     add_definitions(-DCURL_STATICLIB) # Mandatory for building libcurl as static lib
 
     target_include_directories(curl INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/libcurl/libcurl_install/include>)
+        
+    target_link_libraries(curl INTERFACE debug ${CMAKE_CURRENT_BINARY_DIR}/libcurl/libcurl_install/lib/${CURL_DEBUG_TARGET_NAME}${CMAKE_STATIC_LIBRARY_SUFFIX})
```

Comment:
I think the keyword should be changed to PRIVATE as they are not supposed to be exposed in the DLL interface

Reply:
It was built same as libusb (the keyword is not part of this PR), ifI try to change it to private I get lots of cmake errors like:

CMake Error at CMake/external_libcurl.cmake:43 (add_library):
  Cannot find source file:

    PRIVATE

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .m .M .mm .h .hh .h++ .hm

@maloel  / @ev-mp , please say how much effort should be made for changing to PRIVATE.

According to CMAKE , INTERFACE is not like PUBLIC.

> Libraries following INTERFACE are appended to the link interface and are not used for linking <target>.